### DISCUSSION TITLE
Migrating from vanilla RN + community UI packages to Dripsy

### DISCUSSION THREAD

**Discussion by bombillazo** [nandorojo/dripsy#244] [Q&A] [answered]
First, I'm super impressed and excited to try out this library along with Solito and Moti. However, we have a setup using RN + RN Paper + various other RN community packages to build our own UI component library. Our approach is to use atomic design to create cross-platform base elements (using `native.tsx`, `web.tsx` extensions). We import the RN Paper or community components, customize them and return them as custom base components. Fer web we'd do the same with the "web" packages of said components.

I would like to understand how complex it would be to move to Dripsy if we depend on other RN libraries. Would we be required to drop the use of external components and build them out using Dripsy components?

Thanks for the help and for creating such an awesome library!

**Comment by nandorojo** [comment, marked as answer]
It should be simple! As long as your components accept an RN `style` prop, they're compatible with Dripsy. Dripsy is a very simple implementation for theming and styles, and works with any RN component.

**Reply by bombillazo** [reply]
So how do you go about sharing code between web and RN? That part is still unclear to me. Since to be able to reuse feature screens and components across platforms, you can't use components in Next.js that have the RN syntax... can you?

For example, can I use React Native Paper UI components within the Next.js app?

**Reply by nandorojo** [reply]
yes you can, as long as you have React Native Web setup.

**Reply by bombillazo** [reply]
Oooh got it! So in the end we do need RN Web set up. Thanks for the reply.
